### PR TITLE
Dev tools fixes and improvements

### LINKED ISSUE #32226: mvn quarkus:deploy fails on a when quarkus-openshift is added to the project
### Describe the bug

In a project that contains `quarkus-openshift` I am running `mvn quarkus:deploy`. The execution fails with an error about both docker and openshift contianer image extension being present.

### Expected behavior

A single container image extension is available and therefore the command should work.

### Actual behavior

[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:deploy (default-cli) on project demo: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.container.image.deployment.ContainerImageProcessor#publishImageInfo threw an exception: java.lang.IllegalStateException: quarkus-container-image-openshift and quarkus-container-image-docker were detected, at most one container-image extension can be present.
[ERROR] Either remove the unneeded ones, or select one by setting any of the following configuration properties: 'quarkus.container-image.builder=docker', 'quarkus.container-image.builder=openshift'
[ERROR] 	at io.quarkus.container.image.deployment.ContainerImageCapabilitiesUtil.getActiveContainerImageCapability(ContainerImageCapabilitiesUtil.java:40)
[ERROR] 	at io.quarkus.container.image.deployment.ContainerImageProcessor.ensureSingleContainerImageExtension(ContainerImageProcessor.java:112)
[ERROR] 	at io.quarkus.container.image.deployment.ContainerImageProcessor.publishImageInfo(ContainerImageProcessor.java:54)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
[ERROR] 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:833)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
[ERROR] -> [Help 1]

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by iocanel**
The pull request deals with the following:

## maven
- check for available extensions (image & deploy) before falling back to defaults.

## gradle
- check for available extensions (image & deploy) before falling back to defaults

## cli
- fix wrong description for `quarkus deploy kind` command.
- fix commands not picking up project configuration (use `resources:resources`)

### plugins
- fix description for `quarkus plug remove` command.
- latency issues related with cli plugins.
- allows adding directly aliases from remote catalogs (wihtout to install the alias locally).
- introduce java plugin types.
- add more descriptive messages for cli plug commands
- bug fixes.

Resolves: #32226

### CODE REVIEW

**Comment by iocanel** [comment]
This pull request is a superset of #32256 so I will close that and apply requested changes to this one.

**Review by Sgitario — CHANGES_REQUESTED** [review]
I'm not very familiar with the changes in the CLI and Maven/Gradle modules, so will delegate the code review to Max.
However, doing `mvn quarkus:deploy` in a project that uses the OpenShift extension is not equivalent to do `mvn clean install -Dquarkus.kubernetes.deploy=true` (it does not fail, but the binary are not uploaded, and hence the build does not start).

**Comment by iocanel** [comment]
> I'm not very familiar with the changes in the CLI and Maven/Gradle modules, so will delegate the code review to Max. However, doing `mvn quarkus:deploy` in a project that uses the OpenShift extension is not equivalent to do `mvn clean install -Dquarkus.kubernetes.deploy=true` (it does not fail, but the binary are not uploaded, and hence the build does not start).

The two commands are not equivallent. The `quarkus:deploy` does not perform a build by default.

**Comment by Sgitario** [comment]
> > I'm not very familiar with the changes in the CLI and Maven/Gradle modules, so will delegate the code review to Max. However, doing `mvn quarkus:deploy` in a project that uses the OpenShift extension is not equivalent to do `mvn clean install -Dquarkus.kubernetes.deploy=true` (it does not fail, but the binary are not uploaded, and hence the build does not start).
> 
> The two commands are not equivallent. The `quarkus:deploy` does not perform a build by default.

Any technical reason why both commands behave differently?

**Comment by iocanel** [comment]
> > > I'm not very familiar with the changes in the CLI and Maven/Gradle modules, so will delegate the code review to Max. However, doing `mvn quarkus:deploy` in a project that uses the OpenShift extension is not equivalent to do `mvn clean install -Dquarkus.kubernetes.deploy=true` (it does not fail, but the binary are not uploaded, and hence the build does not start).
> > 
> > 
> > The two commands are not equivallent. The `quarkus:deploy` does not perform a build by default.
> 
> Any technical reason why both commands behave differently?

Not a technical decsion. It's more a UX one.

File: devtools/cli/src/main/java/io/quarkus/cli/BuildToolDelegatingCommand.java [hunk 1]
```diff
@@ -123,7 +123,8 @@ public void prepare(BuildToolContext context) {
 
     public void prepareMaven(BuildToolContext context) {
         BuildSystemRunner runner = getRunner(context);
-        BuildSystemRunner.BuildCommandArgs compileArgs = runner.prepareAction("compiler:compile", context.getBuildOptions(),
+        BuildSystemRunner.BuildCommandArgs compileArgs = runner.prepareAction("resources:resources",
+                context.getBuildOptions(),
```
Comment (gsmet) [inline devtools/cli/src/main/java/io/quarkus/cli/BuildToolDelegatingCommand.java:127]: @ia3andy @aloubyansky I'm not sure if this class is related to the plugins only or not. If not, I'd like to have your opinion on this change.

**Review by aloubyansky — APPROVED** [review]

**Comment by iocanel** [comment]
> > > > I'm not very familiar with the changes in the CLI and Maven/Gradle modules, so will delegate the code review to Max. However, doing `mvn quarkus:deploy` in a project that uses the OpenShift extension is not equivalent to do `mvn clean install -Dquarkus.kubernetes.deploy=true` (it does not fail, but the binary are not uploaded, and hence the build does not start).
> > > 
> > > 
> > > The two commands are not equivallent. The `quarkus:deploy` does not perform a build by default.
> > 
> > 
> > Any technical reason why both commands behave differently?
> 
> Not a technical decsion. It's more a UX one.

@Jose given that this change is intended can you please change your review as it currently blocks the merge?

**Review by gsmet — APPROVED** [review]
I'm gonna merge this as it contains a lot of fixes. We can fine tune later.

**Pull request merged**